subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -956,6 +956,17 @@ include::sample[dir="snippets/valueProviders/fileContentsDo/kotlin",files="build
 In general, you should avoid reading files at configuration time, to avoid invalidating configuration cache entries when the file content changes.
 Instead, you can connect the `Provider` returned by link:{javadocPath}/org/gradle/api/provider/ProviderFactory.html#fileContents-org.gradle.api.file.RegularFile-[providers.fileContents()] to task properties.
 
+[[config_cache:requirements:safe_credentials]]
+
+=== Safe credentials
+
+In general, out of precaution, the configuration cache will throw away any data configured for credentials,
+to avoid risking storing plain-text passwords in the cache.
+
+In order to work under the configuration cache, build scripts should not declare inlined credentials (such as `username` and `password`).
+Instead, they should declare only the type of credentials to be used. As an example, see
+the recommendations for safe configuration of repository credentials under <<declaring_repositories.adoc#sec:handling_credentials, Declaring repositories > Handling credentials>>.
```

Comment:
```suggestion
For security reasons, the configuration cache does not store credentials declared inline.

To use credentials in build scripts with the configuration cache, declare credentials with Gradle Properties. To learn more about using credentials with Gradle Properties, check out the example in the
<<declaring_repositories.adoc#sec:handling_credentials,credential handling documentation>>.
```

Suggestion: cut down on some wordiness and specific examples that don't significantly contribute to comprehension of the problem space. Call out the specific solution (gradle properties) early and clearly.